src/quantum/azext_quantum/_validators.py

```diff
@@ -0,0 +1,51 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+import os
+
+from .operations.workspace import WorkspaceInfo
+from .operations.target import TargetInfo
+
+
+def validate_workspace_info(cmd, namespace):
+    """
+    Makes sure all parameters for a workspace are available.
+    """
+    group = getattr(namespace, 'resource_group_name', None)
+    name = getattr(namespace, 'workspace_name', None)
+    ws = WorkspaceInfo(cmd, group, name)
+
+    if not ws.subscription:
+        raise ValueError("Missing subscription argument")
```

Comment:
pls use CLIError

Reply:
If I do, the command help is not displayed.
With `ValueError`:
```
(base) PS C:\repos\azure-quantum> az quantum workspace set
Command group 'quantum' is in preview. It may be changed/removed in a future release.
usage: az quantum workspace set [-h] [--verbose] [--debug]
                                [--only-show-errors]
                                [--output {json,jsonc,yaml,yamlc,table,tsv,none}]
                                [--query JMESPATH]
                                [--subscription _SUBSCRIPTION]
                                [--workspace-name WORKSPACE_NAME]
                                [--resource-group RESOURCE_GROUP_NAME]
az quantum workspace set: error: Missing resource-group argument
```
with `CLIError`:
```
(base) PS C:\repos\azure-quantum> az quantum workspace set
Command group 'quantum' is in preview. It may be changed/removed in a future release.
Missing resource-group argument
```
I do want to show the help. Is there something else I'm missing?

Reply:
Hi @yungezz . Picking up where this conversation was left. We still use 'ValueError' here in order to automatically get help output. This is the reason why @anpaz went this route. Are you ok with this approach, or should we discuss it further?

Reply:
Thanks @ricardo-espinoza.   @houk-ms do you see any issue here?

Reply:
Either one is OK.

@anpaz @ricardo-espinoza 

In the latest versions of AzureCLI Core (>2.12.0), command help won't be printed for either `ValueError` or `CLIError` (https://github.com/Azure/azure-cli/pull/14855, in order to make the CLI error outputs more consistent). So they will be the same if we consider only the output format.

@yungezz 
For telemetry records and error categorizing, both of them are OK.

Reply:
Thanks all!